engine/src/main/java/org/terasology/rendering/dag/StateChange.java

```diff
@@ -22,4 +22,7 @@
     StateChange getDefaultInstance();
 
     RenderPipelineTask generateTask();
+
+    // TODO: investigate a better way?
+    boolean compare(StateChange stateChange);
```

Comment:
I'd recommend to rename this metho to "equals". "Compare" sort of implies that something might be bigger, equal, or smaller than something else.